src/cxx_vector.rs

```diff
@@ -145,6 +157,34 @@ macro_rules! impl_vector_element {
                 }
                 unsafe { __vector_size(v) }
             }
+            fn __vector_slice(v: &CxxVector<$ty>) -> &[$ty] {
+                extern "C" {
+                    attr! {
+                        #[link_name = concat!("cxxbridge04$std$vector$", $segment, "$data")]
+                        fn __vector_data(_: &CxxVector<$ty>) -> &$ty;
+                    }
+                    attr! {
+                        #[link_name = concat!("cxxbridge04$std$vector$", $segment, "$size")]
+                        fn __vector_size(_: &CxxVector<$ty>) -> usize;
+                    }
+                }
+                let data : *const $ty = unsafe { __vector_data(v) };
+                if (data.is_null())
+                {
+                    // Avoid undefined behaviour in slice::from_raw_parts() as
+                    // std::vector::data() returns nullptr if no capacity().
+                    // Alternative: Use from_raw_parts(NonNull::dangling(), 0)
+                    let ret = <&[$ty]>::default(); // empty slice
+                    ret
+                }
+                else
+                {
+                    // std::vector with capacity() has valid data ptr, even if
+                    // size() == 0.
+                    let ret = unsafe { slice::from_raw_parts(data, __vector_size(v)) };
```

Comment:
This is still undefined behavior, I think. As far as I can tell nothing in https://en.cppreference.com/w/cpp/container/vector/data would guarantee that `data` produces a sufficiently aligned pointer in the empty case. Can you find that in the standard, or implement a different way?

Reply:
You are right. Went after the implementation behaviour that always has been instead of reading the standard.

Fix would be to check the size() == 0 instead the ptr, which I see you have done in your implementation. Thanks for fixing my code up. ;)